agency-api-spec.md

```diff
@@ -25,6 +26,37 @@ This draft does not address:
 * size or rate limits
 * error message/status code related to exceeded the rate limit
 * any subsequent calls to the internal FOIA.gov API (to capture info needed to subsequently retrieve status, for example)
+* detailed security controls
+
+
+### Security controls
+
+Once we have confidence in our data models and the touch points for
+interoperability, we will be working with the security team at DOJ to ensure the
+Portal meets the federal requirements for security controls. For now, we include
+only some preliminary ideas based on the agency feedback we’ve received.
+
+Each agency will be responsible for implementing security controls on their own
+end as per their agency regulations and authority to operate. This may include
+configuration of a web application firewall, anti-virus scanning of
+file attachments, and validation of HTTP headers.
+
+
+#### HTTPS
+
+Agency endpoints should be restricted to HTTPS only with valid TLS certificates.
+The Portal will validate your certificate as part of the HTTPS request.
+
+
+#### Authentication
+
+To ensure that your API and case management system aren't publicly writable, we
+recommend restricting your API access to the FOIA.gov Portal. This can be done
+via a shared secret HTTP header token. You will provide this secret token to the
+Portal though configuration. Every request from the Portal will include this
+token, and your API should validate that it is the correct token.
```

Comment:
Did you have an idea of what header we'd be using? I'm guessing a custom one? We either need to include that detail here or as part of any communication that takes place when onboarding agencies. Thoughts?

Reply:
I'm modeling it after how [api.data.gov](https://github.com/18F/api.data.gov/wiki/User-Manual:-Agencies#securing-your-api-backend) does it. Ideally, that means agencies would choose the header they want. Ideally, they could also choose Basic Auth. For this first version we can just choose a hard-coded header (your choice). Agencies could put their API behind api.data.gov and then that would allow them to configure api.data.gov to accept our header, and then api.data.gov could be configured to use a custom header (or even Basic Auth).

So.... Let's pick hard-coded header. How about `X-FOIA-API-SECRET`?

Reply:
Should we drop the `X-`? According to [this stackoverflow post](https://stackoverflow.com/questions/3561381/custom-http-headers-naming-conventions) (so official, I know) the recommendation to include the `X` has been deprecated. It references [RFC 6648](https://tools.ietf.org/html/rfc6648).

So we'd be looking at `FOIA-API-SECRET`

Reply:
Oh, awesome, I hadn't heard about that. Yeah, let's go with `FOIA-API-SECRET` then.